src/Formats/EscapingRuleUtils.cpp

```diff
@@ -193,30 +203,151 @@ void writeStringByEscapingRule(const String & value, WriteBuffer & out, FormatSe
     }
 }
 
-String readStringByEscapingRule(ReadBuffer & buf, FormatSettings::EscapingRule escaping_rule, const FormatSettings & format_settings)
+template <bool read_string>
```

Comment:
Why is it needed?

Reply:
Just to reduce code duplication in `readFieldByEscapingRule` and `readStringByEscapingRule`. If you want to know the difference between these two functions, then `readStringByEscapingRule` reads `String` by escaping rule and return the value of this `String` (ex: for `Quoted` rule we read `'String'` and return `String`), `readFieldByEscapingRule` reads an arbitrary field by escaping rule and return it (ex: for `Quoted` rule we read `'String'` and return `'String'`), so we save all quotes in this function (because we will use them to determine the type).